SmartDeviceLink-iOS/SmartDeviceLink/SDLProxy.m

```diff
@@ -592,6 +621,25 @@ - (NSURLSessionDataTask *)dataTaskForSystemRequestURLString:(NSString *)urlStrin
     return [[NSURLSession sharedSession] dataTaskWithURL:[NSURL URLWithString:urlString]];
 }
 
+/**
+ *  Generate an NSURLSessionDownloadTask for System Requests
+ *
+ *  @param urlString     A string containing the URL to download data from
+ *  @param completionHandler a completion handler returning the response from the server and location of the download
+ *
+ *  @return The download task, which can be started by calling -[resume]
+ */
+- (NSURLSessionDownloadTask *)downloadTaskForSystemRequestURLString:(NSString *)urlString completionHandler:(URLSessionDownloadTaskCompletionHandler)completionHandler {
+    NSParameterAssert(urlString != nil);
+    NSParameterAssert(completionHandler != nil);
+    
+    NSString *logMessage = [NSString stringWithFormat:@"OnSystemRequest (HTTP Request to URL: %@", urlString];
+    [SDLDebugTool logInfo:logMessage withType:SDLDebugType_RPC toOutput:SDLDebugOutput_All toGroup:self.debugConsoleGroupName];
+    
+    // Create and return the download task
+    return [[NSURLSession sharedSession] downloadTaskWithURL:[NSURL URLWithString:urlString] completionHandler:completionHandler];
```

Comment:
@joeljfischer Yes, and Infosys will have a pull request soon that has the NSURLSession / NSURLConnection abstraction, which should work for this as well.